tool/docs/python/PythonDocsParser.kt

```diff
@@ -0,0 +1,190 @@
+    return Class(
+        name = className,
+        description = classDescr,
+        enumConstants = members,
+        examples = classExamples,
+        methods = methods,
+    )
+}
+
+fun parseMethod(element: Element): Method {
+    val methodSignature = enhanceSignature(element.selectFirst("dt.sig-object")!!.text())
+    val methodName = element.selectFirst("dt.sig-object span.sig-name")!!.text()
+    val allArgs = getArgsFromSignature(element.selectFirst("dt.sig-object")!!)
+    val methodReturnType = element.select(".sig-return-typehint").text()
+    val methodDescr = element.select("dl.method > dd > p").map { reformatTextWithCode(it.html()) }
+    val methodArgs = element.select(".field-list > dt:contains(Parameters) + dd p").map {
+        val argName = it.selectFirst("strong")!!.text()
+        assert(allArgs.contains(argName))
+        val argDescr = reformatTextWithCode(removeArgName(it.html())).removePrefix(" – ")
+        Variable(
+            name = argName,
+            defaultValue = allArgs[argName]?.second,
+            description = argDescr,
+            type = allArgs[argName]?.first,
+        )
+    }
+    val methodReturnDescr = element.select(".field-list > dt:contains(Returns) + dd p").text()
+    val methodExamples = element.select("section:contains(Examples) .highlight").map { it.text() }
+
+    return Method(
+        name = methodName,
+        signature = methodSignature,
+        args = methodArgs,
+        description = methodDescr,
+        examples = methodExamples,
+        returnDescription = methodReturnDescr,
+        returnType = methodReturnType,
+    )
+
+}
+
+fun parseProperty(element: Element): Variable {
+    val name = element.selectFirst("dt.sig-object span.sig-name")!!.text()
+    val type = element.selectFirst("dt.sig-object span.sig-name + .property ")?.text()?.dropWhile { !it.isLetter() }
+    val descr = element.select("dd > p").map { reformatTextWithCode(it.html()) }.joinToString("\n\n")
+    return Variable(
+        name = name,
+        description = descr,
+        type = type,
+    )
+}
+
+fun parseEnumConstant(element: Element): EnumConstant {
+    val name = element.selectFirst("dt.sig-object span.sig-name")!!.text()
+    val value = element.selectFirst("dt.sig-object span.sig-name + .property")!!.text().removePrefix("= ")
+    return EnumConstant(
+        name = name,
+        value = value,
+    )
+}
+
+fun getArgsFromSignature(methodSignature: Element): Map<String, Pair<String?, String?>> {
+    return methodSignature.select(".sig-param").map {
+        it.selectFirst(".n")!!.text() to
+                Pair(it.select(".p + .w + .n").text(), it.selectFirst("span.default_value")?.text())
+    }.toMap()
+}
+
+fun reformatTextWithCode(html: String): String {
+    return removeAllTags(replaceEmTags(replaceCodeTags(html)))
+}
+
+fun removeArgName(html: String): String {
+    return Regex("<strong>[^<]*</strong>").replace(html, "")
+}
+
+fun enhanceSignature(signature: String): String {
```

Comment:
is this `enhance`-ing? I think it's actually simplifying right?

Reply:
Maybe... But the signature looks better after this function. It makes some transformations, in fact. For me, `enhance-`, `simplify-` and `transform-` all aren't ideal here. If you definitely prefer `simplify-`, this variant will win.